MC/config/PWGHF/pythia8/generator/pythia8_charmhadronic_with_decays.cfg

```diff
@@ -0,0 +1,107 @@
+### author: Fabrizio Grosa (fabrizio.grosa@cern.ch)
+### since:  January 2022
+
+### beams
+Beams:idA 2212			# proton
+Beams:idB 2212 			# proton
+Beams:eCM 13600. 		# GeV
+
+### processes
+SoftQCD:inelastic on		# all inelastic processes
+
+### decays
+ParticleDecays:limitTau0 on	
+ParticleDecays:tau0Max 10.
+
+### Force golden charm hadrons decay modes for trigger studies
+
+### add D+ decays absent in PYTHIA8 decay table and set BRs from PDG for other
+411:oneChannel = 1 0.0752 0 -321 211 211
+411:addChannel = 1 0.0104 0 -313 211
+411:addChannel = 1 0.0156 0 311 211
+411:addChannel = 1 0.00276 0 333 211
+## add Lc decays absent in PYTHIA8 decay table and set BRs from PDG for other
+4122:oneChannel = 1 0.0196 100 2212 -313
+4122:addChannel = 1 0.0108 100 2224 -321
+4122:addChannel = 1 0.022 100 3124 211
+4122:addChannel = 1 0.035 0 2212 -321 211
+### add Xic+ decays absent in PYTHIA8 decay table
+4232:addChannel = 1 0.2 0 2212 -313
+4232:addChannel = 1 0.2 0 2212 -321 211
+4232:addChannel = 1 0.2 0 3324 211
+4232:addChannel = 1 0.2 0 3312 211 211
+### add Xic0 decays absent in PYTHIA8 decay table
+4132:addChannel = 1 0.2 0 3312 211
+
+### K* -> K pi
+313:onMode = off
+313:onIfAll = 321 211
+### for Ds -> Phi pi+
+333:onMode = off
+333:onIfAll = 321 321
+### for D0 -> rho0 pi+ k-
+113:onMode = off
+113:onIfAll = 211 211
+### for Lambda_c -> Delta++ K-
+2224:onMode = off
+2224:onIfAll = 2212 211
+### for Lambda_c -> Lambda(1520) K-
+102134:onMode = off
+102134:onIfAll = 2212 321
+
+### switch off all decay channels
+411:onMode = off
+421:onMode = off
+431:onMode = off
+4112:onMode = off
+4122:onMode = off
+4232:onMode = off
+4132:onMode = off
+443:onMode = off
+4332:onMode = off
+
+### D0 -> K pi
+421:onIfMatch = 321 211
+
+### D+/- -> K pi pi 
+411:onIfMatch = 321 211 211
+### D+/- -> K* pi
+411:onIfMatch = 313 211
+### D+/- -> phi pi
+411:onIfMatch = 333 211
+
+### D_s -> Phi pi
+431:onIfMatch = 333 211
+
+### Lambda_c -> p K*
+4122:onIfMatch = 2212 313
+### Lambda_c -> Delta K
+4122:onIfMatch = 2224 321
+### Lambda_c -> Lambda(1520) pi
+4122:onIfMatch = 3124 211
+### Lambda_c -> p K pi
+4122:onIfMatch = 2212 321 211
+
+### Lambda_c -> pK0s
+4122:onIfMatch = 2212 311 
+
+### Xic+ -> pK*0
+4232:onIfMatch = 2212 313
+### Xic+ -> p K- pi+
+4232:onIfMatch = 2212 321 211
+### Xic+ -> Xi*0 pi+, Xi*->Xi- pi+
+4232:onIfMatch = 3324 211
+### Xic+ -> Xi- pi+ pi+
+4232:onIfMatch = 3312 211 211
+    
+### Xic0 -> Xi- pi+
+4132:onIfMatch = 3312 211
+
+### Omega_c -> Omega pi
+4332:onIfMatch = 3334 211
```

Comment:
It could be consider to have these decays in a separated config, in case the MC productions for Xic and Omegac will be different (i.e. with a trigger on the hadron species), but it can also be modified later on.